### PR TITLE
Package setup update

### PR DESCRIPTION
**Pull request by Goldziher**
This PR updates the package settings by doing the following:

1. add pre-commit
2. improve the travis setup
3. cleanup and update the md files
4. improve the project folder structure
5. add stubgen scripts
6. update setup.py

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,61 +1,23 @@
...
+  - "3.8"
 
 before_install: |
-  sudo add-apt-repository ppa:ubuntugis/ubuntugis-unstable -y
-  sudo apt-get update
-  sudo apt-get install -y binutils libproj-dev gdal-bin
   pip install -U pip setuptools wheel
 install: |
-  pip install -r ./dev-requirements.txt
+  pip install tox
```
Comment (sobolevn) [inline .travis.yml]: Let's move to Github Actions!

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,8 +1,10 @@
-black
-pytest-mypy-plugins==1.3.0
-psycopg2-binary
-flake8==3.7.9
-flake8-pyi==19.3.0
-isort==4.3.21
-gitpython==3.1.0
+wheel
+mypy==0.782
```
Comment (sobolevn) [inline dev-requirements.txt]: Let's bump `mypy` to the latest version

File: dev-requirements.txt [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+requests==2.24.0
+coreapi==2.3.3
+typing-extensions==3.7.4.3
+gitpython==3.1.9
+pre-commit==2.7.1
+pytest==6.1.1
+pytest-mypy-plugins==1.5.0
```
Comment (sobolevn) [inline dev-requirements.txt]: `1.6.0`

File: setup.cfg [hunk 4]
```diff
@@ -1,19 +1,11 @@
...
-multi_line_output = 5
-wrap_length = 120
-known_first_party = mypy_django_plugin
-
 [flake8]
-exclude =
-  django-sources
-  django-stubs
-  test-data
+exclude = .*/
```
Comment (sobolevn) [inline setup.cfg:2]: The same from `drf-stubs` review

File: setup.py [hunk 5]
```diff
@@ -17,19 +17,15 @@ def find_stub_files(name: str) -> List[str]:
...
-with open("README.md", "r") as f:
+with open("README.md") as f:
     readme = f.read()
 
-dependencies = [
-    "mypy>=0.782,<0.790",
-    "typing-extensions",
-    "django",
-]
+dependencies = ["mypy>=0.782", "typing-extensions>=3.7.2", "django>=2.2.0"]
```
Comment (sobolevn) [inline setup.py:23]: `mypy>=0.790`

File: setup.py [hunk 6]
```diff
@@ -17,19 +17,15 @@ def find_stub_files(name: str) -> List[str]:
...
 
 setup(
     name="django-stubs",
     version="1.6.0",
-    description="Mypy stubs for Django",
+    description="PEP-484 stubs for for Django",
```
Comment (sobolevn) [inline setup.py]: Please, let's leave a description as-is

File: setup.py [hunk 7]
```diff
@@ -42,15 +38,17 @@ def find_stub_files(name: str) -> List[str]:
     packages=["django-stubs", *find_packages(exclude=["scripts"])],
     package_data={"django-stubs": find_stub_files("django-stubs")},
     classifiers=[
-        "Development Status :: 3 - Alpha",
         "License :: OSI Approved :: MIT License",
+        "Operating System :: OS Independent",
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
```
Comment (sobolevn) [inline setup.py]: Do we actually test on 3.9?

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot! 👍

File: setup.py (hunk 5 above)
Comment (Goldziher) [reply setup.py:23]: are you sure we want to restrict like this? I'd say 0.750 and above, we dont want to restrict people

File: setup.cfg (hunk 4 above)
Comment (Goldziher) [reply setup.cfg:2]: kk

File: .travis.yml (hunk 1 above)
Comment (Goldziher) [reply .travis.yml]: Lets do this in a separate PR :)

File: dev-requirements.txt (hunk 2 above)
Comment (Goldziher) [reply dev-requirements.txt]: done

**Comment by Goldziher** [comment]
replaced by rebased version

**Pull request closed without merging**